src/agency-budget/agency-budget.repository.ts

```diff
@@ -21,9 +24,21 @@ export class AgencyBudgetRepository {
     })
     .prepare("checkByCode");
 
-  async checkByCode(code: string): Promise<CheckByCodeRepo | undefined> {
+  async checkByCode(code: string): Promise<CheckByCodeRepo> {
+    const key = JSON.stringify({
+      code,
+      domain: "agencyBudget",
+      function: "checkByCode",
+    });
+
+    const cachedValue: boolean | null = await this.cacheManager.get(key);
```

Comment:
Just to sanity check my understanding of this code, when you do `cacheManager.get`, that function checks to see if a cached value is stale and returns null if it is? Or does that happen somewhere else?

Reply:
Correct, that is what happens:
[link to docs](https://www.npmjs.com/package/cache-manager#get)